src/cmd/support_archive/resource_query.go

```diff
@@ -0,0 +1,100 @@
+package support_archive
+
+import (
+	"reflect"
+
+	"github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
+	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects"
+	"github.com/Dynatrace/dynatrace-operator/src/webhook"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type resourceQueryGroup struct {
+	resources []schema.GroupVersionKind
+	filters   []client.ListOption
+}
+
+type resourceQuery struct {
+	groupVersionKind schema.GroupVersionKind
+	filters          []client.ListOption
+}
+
+func getQueries(namespace string) []resourceQuery {
+	allQueries := make([]resourceQuery, 0)
+	allQueries = append(allQueries, getInjectedNamespaceQueryGroup().getQueries()...)
+	allQueries = append(allQueries, getOperatorNamespaceQueryGroup(namespace).getQueries()...)
+	allQueries = append(allQueries, getOperatorComponentsQueryGroup(namespace).getQueries()...)
+	return allQueries
+}
+
+func getInjectedNamespaceQueryGroup() resourceQueryGroup {
+	return resourceQueryGroup{
+		resources: []schema.GroupVersionKind{
+			toGroupVersionKind(corev1.SchemeGroupVersion, corev1.Namespace{}),
+		},
+		filters: []client.ListOption{
+			client.HasLabels{
+				webhook.InjectionInstanceLabel,
+			},
+		},
+	}
+}
+
+func getOperatorNamespaceQueryGroup(namespace string) resourceQueryGroup {
+	return resourceQueryGroup{
+		resources: []schema.GroupVersionKind{
+			toGroupVersionKind(corev1.SchemeGroupVersion, corev1.Namespace{}),
+		},
+		filters: []client.ListOption{
+			client.MatchingFields{
+				"metadata.name": namespace,
+			},
+		},
+	}
+}
+
+func getOperatorComponentsQueryGroup(namespace string) resourceQueryGroup {
+	return resourceQueryGroup{
+		resources: []schema.GroupVersionKind{
+			toGroupVersionKind(appsv1.SchemeGroupVersion, appsv1.Deployment{}),
+			toGroupVersionKind(appsv1.SchemeGroupVersion, appsv1.StatefulSet{}),
+			toGroupVersionKind(appsv1.SchemeGroupVersion, appsv1.DaemonSet{}),
+			toGroupVersionKind(appsv1.SchemeGroupVersion, appsv1.ReplicaSet{}),
+			toGroupVersionKind(corev1.SchemeGroupVersion, corev1.Service{}),
+			toGroupVersionKind(corev1.SchemeGroupVersion, corev1.Pod{}),
+			toGroupVersionKind(v1beta1.GroupVersion, v1beta1.DynaKube{}),
```

Comment:
nitpick:
this list feels like a limited selection of what we could collect.
probably every single component could be collected that comes with the install, but these are the most interesting ones so its fine for now.
stuff like `MutatingWebhookConfiguration`, `ValidatingWebhookConfiguration` could be useful, but they have certs, even if generated by us, it still feels wrong collecting them.